posts/2022-03-01-user-feature-install.adoc

```diff
@@ -0,0 +1,378 @@
+
+
+
+Run `**_mvn clean install_**` to create an ESA file. Inside the ESA file, you will find your bundle JAR and SUBSYSTEM.MF.
+
+.SUBSYSTEM.MF
+[source, txt]
+----
+Subsystem-ManifestVersion: 1
+Subsystem-SymbolicName: com.example.bundle.Activator;visibility:=public
+Subsystem-Version: 1.0.0
+Subsystem-Name: SimpleActivatorESA
+Subsystem-Content: com.example.bundle.Activator;version="[1.0.0,1.0.0]"
+IBM-Feature-Version: 2
+IBM-ShortName: SimpleActivator
+Subsystem-Type: osgi.subsystem.feature
+Subsystem-Vendor: IBM
+----
+'''
+=== Creating Bill of Materials (BOM) 
+Create a Bill of Materials (BOM) for the user feature ESA file. The BOM is a pom file that manages the dependencies of the project. Liberty Maven Plugin's `prepare-feature` goal and `install-feature` goal require the BOM file to install the Liberty user feature onto Liberty runtime. 
+[source, xml]
+.pom.xml
+----
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>test.user.test.osgi</groupId>
+  <artifactId>features-bom</artifactId>
+  <version>0.0.1-SNAPSHOT</version>
+  <packaging>pom</packaging>
+  <name>user features bill of materials</name>
+  <description>user features bill of materials</description>
+  <url>https://openliberty.io/</url>
+  
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>test.user.test.osgi</groupId>
+        <artifactId>SimpleActivatorESA-</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+        <scope>runtime</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+</project>
+----
+Run `**_mvn clean install_**` to create `features-bom.pom` BOM file.
+
+'''
+=== Installing Liberty user feature using Liberty Maven Plugin
+
+[start=1]
+. To enable link:https://github.com/OpenLiberty/ci.maven[Liberty Maven Plugin] in your project, add the following to your pom.xml
++
+----
+<plugin>
+   <groupId>io.openliberty.tools</groupId>
+   <artifactId>liberty-maven-plugin</artifactId>
+   <!-- Specify configuration, executions for liberty-maven-plugin -->
+   <configuration>
+      <serverName>test</serverName>
+   </configuration>
+</plugin>
+----
+
+. Run `**_mvn liberty:create_**` to create a Liberty server `test`
+
+. Specify the feature to install for the `test` server.
++
+Open `${project.build.testOutputDirectory}/wlp/server.xml` file and add the user feature. The extension `usr` indicates that the feature will be installed to `${project.build.testOutputDirectory}/wlp/usr` directory. It also tells the server where to find the user feature when the server starts. 
```

Comment:
The extension `usr` indicates that the feature will be installed to `${project.build.testOutputDirectory}/wlp/usr` directory.

--->

The `usr` extension indicates that the feature will be installed to  the `${project.build.testOutputDirectory}/wlp/usr` directory.